### PR TITLE
Replace pad_token with -100 for LM loss calculation

### PR DESCRIPTION
**Pull request by setu4993**
The docs for both GPT and [GPT2](https://huggingface.co/transformers/model_doc/gpt2.html#gpt2lmheadmodel) specify that labels that are not -100 will be used for the calculation of the loss. So, the padding for the labels should be `-100`, not `tokenizer.pad_token_id`.

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
This is correct! Thanks @setu4993!

**Review by julien-c — APPROVED** [review]

**Comment by setu4993** [comment]
Closed the PR by mistake... Re-opening it.

**Comment by julien-c** [comment]
Just a quick question for @mfuntowicz, is `copy.deepcopy` a performant way to clone a tensor? (given that this is called at each training step)

**Comment by setu4993** [comment]
I saw `deepcopy` being used elsewhere in the code, so added it as is. Just looking at the new tensor [documentation](https://pytorch.org/docs/stable/tensors.html#torch.Tensor.new_tensor), and they recommend using `.clone().detach()`. Happy to change it to that.

**Comment by setu4993** [comment]
Bumping this since I haven't seen any activity in a few days.

**Comment by julien-c** [comment]
Yes `.clone().detach()` sounds good.

**Review by mfuntowicz — APPROVED** [review]

**Comment by julien-c** [comment]
LGTM but let's let @LysandreJik have a last check and merge this

**Comment by setu4993** [comment]
Thanks @julien-c!

**Comment by setu4993** [comment]
Hey @LysandreJik, can you please review when you have a chance? Thanks!

**Review by LysandreJik — APPROVED** [review]

**Comment by LysandreJik** [comment]
Thanks @setu4993!
